app/src/main/java/org/jellyfin/androidtv/model/trailers/lifter/ExternalTrailerLifter.kt

```diff
@@ -1,15 +1,28 @@
 package org.jellyfin.androidtv.model.trailers.lifter
 
-import android.net.Uri
 import org.jellyfin.androidtv.model.trailers.external.ExternalTrailer
 import org.jellyfin.apiclient.model.entities.MediaUrl
+import java.net.MalformedURLException
+import java.net.URL
 
 abstract class ExternalTrailerLifter {
 	private val domainRegex = Regex("""^(.*\.|)(.+\...+)$""")
 
-	protected fun getDomain(uri: Uri): String? = uri.host?.let { host -> domainRegex.find(host)?.groups?.get(2)?.value }
+	protected fun getDomain(url: URL): String? {
+		return url.host?.let { host -> domainRegex.find(host)?.groups?.get(2)?.value }
```

Comment:
I've once read that the URI class is preferred above URL because URL will make DNS queries but I can't find a proper source for it right now.

Reply:
That would be really unexpected, but idk. Maybe I should try to wireshark the emulator.

Reply:
Using the Android URI class will give you some additional methods like getting a query parameter too (to replace the `v=xxx` regex):
https://developer.android.com/reference/android/net/Uri